explore/explore_internal.h

```diff
@@ -90,7 +90,8 @@ namespace exploration
       return E_EXPLORATION_BAD_RANGE;
 
     float norm = 0.;
-    float max_score = *std::max_element(scores_first, scores_last);
+    float max_score = lambda > 0 ? *std::max_element(scores_first, scores_last)
```

Comment:
Can you explain the reasoning behind the change? Why does it make sense to find the min element for max score?

Reply:
The reasoning is: 
* Right now it doesn't work.  (See new test number 200; inspired by a problem reported by a customer).
* This routine was written assuming lambda was positive but it is actually negative in practice ("minimizing cost" instead of "maximizing reward").
* To caller-proof this routine I decided to make it handle any lambda properly.

Reply:
Sounds good.